cpp/src/c_api/pagerank.cpp

```diff
@@ -16,77 +16,287 @@
+            handle_, graph_, error_.get());
+        if (error_code_ != CUGRAPH_SUCCESS) return;
+      }
+
+      auto graph = reinterpret_cast<cugraph::graph_t<vertex_t, edge_t, weight_t, true, multi_gpu>*>(
+        graph_->graph_);
+
+      auto graph_view = graph->view();
+
+      auto number_map = reinterpret_cast<rmm::device_uvector<vertex_t>*>(graph_->number_map_);
+
+      rmm::device_uvector<vertex_t> vertex_ids(graph->get_number_of_vertices(),
+                                               handle_.get_stream());
+      rmm::device_uvector<weight_t> pageranks(graph->get_number_of_vertices(),
+                                              handle_.get_stream());
+
+      if (personalization_vertices_ != nullptr) {
+        //
+        // Need to renumber personalization_vertices
+        //
+        renumber_ext_vertices<vertex_t, multi_gpu>(handle_,
+                                                   personalization_vertices_->as_type<vertex_t>(),
+                                                   personalization_vertices_->size_,
+                                                   number_map->data(),
+                                                   graph_view.get_local_vertex_first(),
+                                                   graph_view.get_local_vertex_last(),
+                                                   do_expensive_check_);
+      }
+
+      cugraph::pagerank<vertex_t, edge_t, weight_t, weight_t, multi_gpu>(
+        handle_,
+        graph_view,
+        precomputed_vertex_out_weight_sums_
+          ? std::make_optional(precomputed_vertex_out_weight_sums_->as_type<weight_t const>())
+          : std::nullopt,
+        personalization_vertices_
+          ? std::make_optional(personalization_vertices_->as_type<vertex_t const>())
+          : std::nullopt,
+        personalization_values_
+          ? std::make_optional(personalization_values_->as_type<weight_t const>())
+          : std::nullopt,
+        personalization_vertices_
+          ? std::make_optional(static_cast<vertex_t>(personalization_vertices_->size_))
+          : std::nullopt,
+        pageranks.data(),
+        static_cast<weight_t>(alpha_),
+        static_cast<weight_t>(epsilon_),
+        max_iterations_,
+        has_initial_guess_,
+        do_expensive_check_);
+
+      cugraph::detail::sequence_fill(
+        handle_.get_stream(), vertex_ids.data(), vertex_ids.size(), vertex_t{0});
+
+      std::vector<vertex_t> vertex_partition_lasts = graph_view.get_vertex_partition_lasts();
+
+      unrenumber_int_vertices<vertex_t, multi_gpu>(handle_,
+                                                   vertex_ids.data(),
+                                                   vertex_ids.size(),
+                                                   number_map->data(),
+                                                   vertex_partition_lasts,
+                                                   do_expensive_check_);
```

Comment:
Same here, is this necessary? Can't we just copy number_map->data() instead?

Reply:
Will resolve same was as above.

Reply:
So, I guess this hasn't been addressed and will be addressed in a future commit?

Reply:
Sorry, please this regard, I assume you will address this in https://github.com/rapidsai/cugraph/issues/1947

Reply:
I believe I addressed the immediate change here (does a copy now).  #1947 will address the longer-term question of preserving the number_map and including it in the results rather than copying.